쏘카 데이터 엔지니어링 팀 신입 디니의 온보딩 후기 (#75)
socar-inc/socar-inc.github.io: _posts/2021-12-02-data-engineering-team-onboarding.md

이 과제에서도 웹 프레임워크는 딱히 정해지지 않았지만, 저희 팀에서 FastAPI를 자주 쓰고 있었기 때문에 겸사겸사 해서 FastAPI로 웹서버를 구현했습니다. (이 과정에서 저만의 셀프 FastAPI 온보딩 과제- 간단한 CRUD 서버 구축해보기도 있었습니다.)

Helm chart를 직접 만들고 `values.yaml` 을 작성하는 방법, GKE 환경을 configure 하는 방법 등 짧은 시간에 많은 걸 접하고 배울 수 있었습니다. 처음에는 Kubernetes의 개념을 익히기 위해 minikube 로 테스트를 해봤는데요. 나중에는 저희 팀 GKE 에서만 되는 설정들(GCR에 있는 이미지를 끌어오기, ingress 할당 등)이 있어서 조금 해맸던 기억이 있습니다.  

<br>

### 과제 후 얻은 것

이렇게 4개의 과제를 완료하는데 총 2주가 걸렸는데요. 신입의 입장에서, 이렇게 전체적으로 업무의 틀을 파악하는 시간이 주어진게 정말 감사한 일이었습니다. 👍   
모든 회사에서 이런 기회가 주어지지 않는다는 것을 알기에 더욱 소중한 시간이었습니다.

가장 큰 장점은 "업무 적응에 대한 심적 부담이 크게 줄었다!"입니다. 사실 데이터 엔지니어링 팀에 필요한 도메인이 매우 넒은데 비해서 저는 관련 경험이 거의 없어서 처음에 막연한 두려움이 있었어요. 그런데 Task 자체는 매우 단순화한 상태에서 프레임워크를 사용해보고 플로우를 익혀보니, 좀더 복잡한 업무도 "아, 일단 이건 해봤으니까 여기서 발전해나가면 되겠구나!"하는 자신감이 생겼습니다.

첫 환경 세팅이나 배포의 난관을 온보딩 과제를 통해 극복할 수 있던 것도 큰 의미였습니다. 프레임워크 뿐만 아니라 [Lens](https://k8slens.dev/) 등 팀에서 활용하고 있는 모니터링 툴도 이때 빨리 접할 수 있었구요. 팀 문서나 코드도 점점 눈에 들어오기 시작했습니다. 그리고 트러블슈팅 과정을 기록한 것들을 공유하며, 제가 어떤 부분에서 부족한지 팀원들의 피드백을 받아볼 수 있어서 좋았습니다.

<br>

### 마무리 발표

이렇게 2주동안 과제를 수행한 뒤, 팀장 토마스의 제안으로 온보딩 과제를 회고하는 발표를 하게 되었습니다. 주로 온보딩 과제와 트러블 슈팅 내용들, 제가 느꼈던 감정 (..) 들 위주였습니다. 발표 후, 이런 식으로 온보딩 과정을 발전 및 정착시켰으면 좋겠다는 논의도 나왔어요.

> ![온보딩 발표 중 Airflow](/img/data-engineering-onboarding/onboarding-pitch-Airflow.png)

Author: 아, 이거 아마 하디 리뷰 과정중 airflow -> Airflow 텍스트 대치를 해서 파일명이 깨진 것 같습니다. 수정하겠습니다.

<br>

---

## 4. 앞으로는 무엇을? <a name="throwback"></a>

많은 분들의 도움이 있었던 온보딩 기간을 거치고, 저는 앞으로 회사에서 이런 것을 하고 싶습니다.

<br>

### 여러 사람이 편해지는 시스템이나 툴을 만들고 싶어요.

데이터 엔지니어링 업무 자체가 서포팅의 성격도 있는데요. 회사에서 저 뿐만이 아니라 여러 사람이 편해지는 시스템이나 툴을 만들고 싶습니다.  
예를 들면 "디니의 트러블슈팅 DB"를 만들고 있는데요. 지금 트러블슈팅 한 과정을 미래의 나 혹은 다른사람이 구글링처럼 편하게 검색하고 찾을 수 있었으면 좋겠다는 생각에서 시작되었습니다.

그리고, 제가 온보딩 과정에서 도움을 많이 받은 만큼 다음 오시는 분을 위해 온보딩 과정을 더욱 발전시키고 싶어요. 개인적인 경험으로는 argoCD 를 통한 배포와, python 협업 환경(테스트 코드 짜기, 디버깅 하기 등)에 대한 온보딩 등이 추가되면 더 좋겠다고 느꼈습니다.  

<br>

### 문화 개선에 기여하고 싶어요.

다같이 일하기 즐거운 회사가 되면 좋겠다는 소망이 있고, 제가 할 수 있는 것부터 하려고 노력 중입니다.
예를 들면 데이터 그룹 해피아워, 워크샵 등 밍글링 행사의 기획을 맡고 있고, "코딩 안풀릴때 소리지르는 방" 슬랙 채널 개설해서 다들 일하다 마음껏 소리지르는 (...) 공간을 만들었어요. 

![코딩 안될때 소리지르는 짤](/img/data-engineering-onboarding/screaming.png)
*입사 이후 최대의 업적 : 코딩 안될때 소리지르는 방 만든 것.*

<br>

### 많이 공유하고 싶어요.

제가 취업 준비 할때 쏘카 테크블로그를 많이 참여하기도 했고, 공부하면서 언제나 다른 사람의 글을 참고하고 있기 때문에, 항상 유용한 글을 쓰고 싶다는 마음이 있습니다. 곧 메타데이터 플랫폼 글로 돌아오겠습니다. 😏

<br>

---

## 5. Q & A <a name="qna"></a>
마지막으로 제가 취업을 준비하면서 스스로 궁금했던, 그리고 비슷한 과정에 계실 분들이 궁금해할만한 질문들과 이에 대한 답변을 정리해보았습니다. 

<br>

### 실무에서 데이터 분석가 or 사이언티스트 VS 데이터 엔지니어의 차이는?

> 데이터 분석가와 데이터 사이언티스트 분들이 맘껏 능력발휘 할수 있는 탄탄한 플레이그라운드를 만드는게 데이터 엔지니어링의 역할인것 같습니다. 제가 취업 준비할 때는 데이터 관련 직군 간의 업무 차이가 잘 와닿지 않았는데요. 실무를 경험해보니 담당하는 업무가 확연히 다른 것 같습니다. (물론 회사마다 정의가 다르고, 작은 규모에서는 같이 하시는 분들도 있지만요.)

Reviewer: ```suggestion
데이터 분석가와 데이터 사이언티스트 분들이 맘껏 능력 발휘할 수 있는 탄탄한 플레이그라운드를 만드는 것이 데이터 엔지니어링의 역할인 것 같습니다. 취업 준비할 때는 데이터 관련 직군 간의 업무 차이가 잘 와닿지 않았는데, 실무를 경험해보니 담당하는 업무가 확연히 다른 것 같습니다. (물론 회사마다 정의가 다르고, 작은 규모에서는 같이 하시는 분들도 있을 것 같습니다)
```

> 아주 단순하게 얘기하자면 데이터 분석가는 말그대로 '분석가', 데이터 엔지니어는 '개발자'의 모습에 가깝다고 생각합니다. 저도 아직 신입이라 섣불리 말하기가 조심스럽지만, 인사이트 뽑아내고 사업적인 고민하는게 좋다면 데이터 분석을, 시스템 구축과 자동화, 프로그래밍 자체에 관심이 많다면 데이터 엔지니어링이 좀더 맞지 않을까 싶습니다. 

Reviewer: ```suggestion
아주 단순하게 얘기하자면 데이터 분석가는 말 그대로 '분석가', 데이터 엔지니어는 '개발자'의 모습에 가깝다고 생각합니다. 어떤 목적을 해결하기 위해 데이터를 통해 분석하거나 사업적인 고민하는 것이 좋다면 데이터 분석을, 시스템 구축과 자동화, 프로그래밍 자체에 관심이 많다면 데이터 엔지니어링이 더 맞지 않을까 싶습니다. 
```

<br>

### 재택근무는 어떤가요?

> 

Reviewer: 재택근무 파트는 갑자기 좀 맥락이 끊기는 느낌이 드네요. 취업 관련 이야기 하다가 재택근무 이야기 나와서 삭제해도 괜찮을듯 합니당

Author: 넵~!

> 쏘카는 현재 자유롭게 재택근무를 할 수 있습니다. 오전 재택, 오후 출근 등의 형태도 가능합니다. 

Reviewer: ```suggestion
쏘카는 현재 자유롭게 재택근무를 할 수 있습니다. 
```

> 여담으로 저는 재택근무에 큰 환상이 있었는데요. 막상 해보니 재택근무도 성향이라는 것을 느꼈습니다. 아무래도 자기통제 잘되는 사람한테 정말 좋은데, 저 같이 자기통제력이 약한 사람에게는 출근하는게 더 잘 맞는것 같아요 😂

Reviewer: ```suggestion
여담으로 저는 재택근무에 큰 환상이 있었습니다. 막상 해보니 재택근무도 성향이라는 것을 느꼈습니다. 아무래도 자기통제 잘되는 사람한테 정말 좋은데, 저 같이 자기통제력이 약한 사람에게는 출근하는게 더 잘 맞는것 같습니다 😂
```

<br>

### 비전공자, 문과라는 백그라운드가 회사에서 어떻게 작용하는지?

> 저도 이 점에 대해서 걱정을 했는데, 저희 팀에는 오히려 비전공자가 더 많고 중요한 건 아무도 과(와 학교)에 신경을 쓰지 않습니다. 그런 백그라운드 보다는 요구하는 일을 할 수 있는 실력이 더 중요하다고 느꼈습니다.

Reviewer: ```suggestion
저도 이 점에 대해서 걱정했는데, 저희 팀에는 오히려 비전공자가 더 많고 중요한 건 아무도 과(와 학교)에 신경을 쓰지 않습니다. 그런 백그라운드보다 요구하는 일을 할 수 있는 실력이 더 중요하다고 느꼈습니다.
```

> 그리고 커뮤니케이션 능력은 어디서 무슨 일을 하든 무조건 플러스라고 생각합니다. 저는 막연하게 개발자라면 코딩만 한다, 사람이랑 얘기 잘 안한다라는 편견을 갖고 있었는데요. 회사에 와서 보니 (물론 실력은 기본이겠지만) 의사결정, 우선순위 산정 등 다른 게 더 중요할 수도 있다는 생각이 들었습니다. [배달의 민족 CEO 인터뷰](https://eoeoeo.net/2021/08/12/%EB%B0%B0%EB%8B%AC%EC%9D%98%EB%AF%BC%EC%A1%B1-ceo%EA%B0%80-%EB%A7%90%ED%95%98%EB%8A%94-%ED%95%A8%EA%BB%98-%EC%9D%BC%ED%95%98%EA%B3%A0-%EC%8B%B6%EC%9D%80-%EA%B0%9C%EB%B0%9C%EC%9E%90%EC%9D%98-%EA%B8%B0/)에서 "개발자는 비즈니스 문제를 해결하는 사람"으로 정의하길 바란다는 말씀을 하시는데요. 이와 같이 코딩 뿐만 아니라 문제 해결에 지치지 않고 재밌어 한다면 개발자가 잘 맞을 것 같습니다.

Reviewer: ```suggestion
그리고 커뮤니케이션 능력은 어디서 무슨 일을 하든 무조건 플러스라고 생각합니다. 회사에 와서 보니 의사결정, 우선순위 산정 등 다른 게 더 중요할 수도 있다는 생각이 들었습니다. [배달의 민족 CEO 인터뷰](https://eoeoeo.net/2021/08/12/%EB%B0%B0%EB%8B%AC%EC%9D%98%EB%AF%BC%EC%A1%B1-ceo%EA%B0%80-%EB%A7%90%ED%95%98%EB%8A%94-%ED%95%A8%EA%BB%98-%EC%9D%BC%ED%95%98%EA%B3%A0-%EC%8B%B6%EC%9D%80-%EA%B0%9C%EB%B0%9C%EC%9E%90%EC%9D%98-%EA%B8%B0/)에서 "개발자는 비즈니스 문제를 해결하는 사람"으로 정의하길 바란다는 말씀을 합니다. 이와 같이 코딩 뿐만 아니라 문제 해결에 지치지 않고 재밌어한다면 개발자가 잘 맞을 것 같습니다.
```

<br>

> ### 신입 데이터 엔지니어를 희망한다면 어떻게 포트폴리오를 꾸리면 좋을지?

Reviewer: ```suggestion
### 신입 데이터 엔지니어를 희망한다면 어떻게 포트폴리오를 꾸리면 좋을까?
```

> 개인적으로는 데이터 엔지니어링이 참 포트폴리오를 준비하기 힘든 분야라고 느꼈는데요. 소소하지만 저의 팁을 공유합니다.

Reviewer: ```suggestion
개인적으로는 데이터 엔지니어링이 참 포트폴리오를 준비하기 힘든 분야라고 느꼈습니다. 소소하지만 저의 팁을 공유합니다.
```

<br>

> #### 1) 가고싶은 회사가 어떤 환경인지 보고 공부하자.

Reviewer: ```suggestion
#### 1) 가고 싶은 회사가 어떤 환경인지 보고 공부하자.
```

> 가고 싶은 회사의 채용공고를 꼼꼼히 읽고, 어떤 툴과 프레임워크를 쓰는지 보시면서 그 프레임워크에 대한 공부를 하시면 좋을 것 같아요.  

Reviewer: ```suggestion
가고 싶은 회사의 채용공고를 꼼꼼히 읽고, 어떤 툴과 프레임워크를 쓰는지 보시면서 그 프레임워크에 대한 공부를 하시면 좋을 것 같습니다

```

> 저희가 알고 있는 IT 기업들은 AWS, GCP, Hadoop 이 선에서 크게 달라지지 않는다고 생각합니다. 온라인 강의 사이트에서 이런 프레임워크를 타겟으로 한 강의도 많아서, 참고하시면 좋을 것 같습니다. 

Reviewer: ```suggestion
저희가 알고 있는 IT 기업들은 AWS, GCP, Hadoop 이 선에서 크게 달라지지 않는다고 생각합니다. 온라인 강의 사이트에서 이런 프레임워크를 타겟으로 한 강의를 참고하시면 좋을 것 같습니다. 
```

> 쏘카 데이터그룹은 기본적으로 GCP 를 이용하고 있기 때문에, 저는 개인적으로 [Qwiklabs](https://www.qwiklabs.com/) 플랫폼의 GCP 관련 실습 강의(ex.[Engineer Data in Google Cloud](https://www.qwiklabs.com/quests/132?catalog_rank=%7B%22rank%22%3A2%2C%22num_filters%22%3A0%2C%22has_search%22%3Atrue%7D&search_id=14949926))가 많은 도움이 되었습니다. Hadoop의 경우 조금 오래된 강의기는 하지만 [Udemy](https://www.udemy.com/) 플랫폼의 [The Ultimate Hands-On Hadoop](https://www.udemy.com/course/the-ultimate-hands-on-hadoop-tame-your-big-data/) 강의를 통해 기본 개념을 익힐 수 있었던 기억이 납니다. 이 외에도 [Udacity](https://www.udacity.com/), [O'Reilly](https://www.oreilly.com/) 등의 강의 플랫폼에서 유용한 강의들을 찾으실 수 있습니다. 

Reviewer: ```suggestion
쏘카 데이터 그룹은 기본적으로 GCP(Google Cloud Platform)를 이용하고 있기 때문에, 개인적으로 [Qwiklabs](https://www.qwiklabs.com/) 플랫폼의 GCP 관련 실습 강의(ex.[Engineer Data in Google Cloud](https://www.qwiklabs.com/quests/132))가 많은 도움되었습니다. Hadoop의 경우 조금 오래된 강의지만 [Udemy](https://www.udemy.com/) 플랫폼의 [The Ultimate Hands-On Hadoop](https://www.udemy.com/course/the-ultimate-hands-on-hadoop-tame-your-big-data/) 강의를 통해 기본 개념을 익힐 수 있었던 기억이 납니다. 이 외에도 [Udacity](https://www.udacity.com/), [O'Reilly](https://www.oreilly.com/) 등의 강의 플랫폼에서 유용한 강의들을 찾으실 수 있습니다. 
```

<br>

#### 2) 나만의 작고 귀여운 ETL 파이프라인을 만들어보자.

> 데이터 엔지니어링 분야에서 가장 무난하게 포트폴리오를 만들 수 있는게 ETL 파이프라인이지 않을까 싶습니다. 꼭 가고싶은 회사의 프레임워크와 일치하지 않아도 상관없으니, 관심있는 API 데이터를 ETL 하는 파이프라인을 만들어보면 거기서부터 관련된 아이디어를 발전시킬 수 있을 거라 생각합니다.

Reviewer: ```suggestion
데이터 엔지니어링 분야에서 가장 무난하게 포트폴리오를 만들 수 있는 것은 ETL 파이프라인이라고 생각합니다. 꼭 가고 싶은 회사의 프레임워크와 일치하지 않아도 상관없으니, 관심 있는 API 데이터를 ETL 하는 파이프라인을 만들어보면 관련된 아이디어를 발전시킬 수 있을 거라 생각합니다.
```

<br>

#### 3) 블로그 잘 관리하기

> 신입 입장에서 이력서로 엄청난 차이를 보여주기는 쉽지 않다고 생각해요. 이럴때, 본인의 강점과 열정을 보여줄 수 있는게 개인 블로그나 포트폴리오 사이트라고 생각합니다. 글이 완벽하지 않더라도 공부한 것들이나 관심있는 내용을 꾸준히 업로드 하시면 좋은 것 같습니다.

Reviewer: ```suggestion
신입 입장에서 이력서로 엄청난 차이를 보여주기는 쉽지 않다고 생각합니다. 이럴 때, 본인을 설명해줄 수 있는 개인 블로그나 포트폴리오 사이트가 있으면 좋습니다. 글이 완벽하지 않더라도 공부한 것들이나 관심있는 내용을 작성하면 좋은 것 같습니다.
```

<br>

> 여기까지 데이터 엔지니어링팀 디니의 4개월 신입 회고였습니다.  온보딩 과정을 도와주신 팀원분들 이자리를 빌어 다시한번 감사드립니다. 

Reviewer: ```suggestion
여기까지 데이터 엔지니어링팀 디니의 4개월 신입 회고였습니다.  온보딩 과정을 도와주신 팀원분들 이 자리를 빌어 다시 한번 감사드립니다. 
```

> 긴 글 읽어주셔서 감사합니다. 궁금한 점이 있으시면 언제든 댓글이나 메세지 주세요 :)

Reviewer: ```suggestion
긴 글 읽어주셔서 감사합니다. 궁금한 점이 있으시면 언제든 댓글 남겨주세요 :)
```
